### PR TITLE
[5.0] HHH-10545, HHH-10546 InfinispanRegionFactory fixes

### PR DESCRIPTION
**Pull request by pferraro**
HHH-10545 InfinispanRegionFactory inadvertently starts the default cache during stop()
https://hibernate.atlassian.net/browse/HHH-10545
HHH-10546 InfinispanRegionFactory should stop any caches it starts
https://hibernate.atlassian.net/browse/HHH-10546

### CODE REVIEW

**Comment by scottmarlow** [comment]
hmm, this change doesn't build for me:
hibernate5/hibernate-infinispan/src/main/java/org/hibernate/cache/infinispan/InfinispanRegionFactory.java:241: error: diamond operator is not supported in -source 1.6
        private List<BaseRegion> regions = new ArrayList<>();
                                                         ^
  (use -source 7 or higher to enable diamond operator)
/mnt/ssd/work/hibernate5/hibernate-infinispan/src/main/java/org/hibernate/cache/infinispan/InfinispanRegionFactory.java:435: error: lambda expressions are not supported in -source 1.6
                regions.forEach( region -> {
                                        ^
  (use -source 8 or higher to enable lambda expressions)
/mnt/ssd/work/hibernate5/hibernate-infinispan/src/main/java/org/hibernate/cache/infinispan/util/CacheCommandFactory.java:52: error: lambda expressions are not supported in -source 1.6
                regions.forEach( region -> allRegions.remove( region.getName() ) );
                                        ^
  (use -source 8 or higher to enable lambda expressions)
3 errors
:hibernate-infinispan:compileJava FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':hibernate-infinispan:compileJava'.
> Compilation failed; see the compiler error output for details.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

BUILD FAILED

**Comment by scottmarlow** [comment]
Another problem is that I don't see the EmbeddedCacheManager.undefineConfiguration in Infinispan 7.2.1.Final

**Comment by scottmarlow** [comment]
EmbeddedCacheManager.undefineConfiguration(String) was added to Infinispan 8.0 which Hibernate 5.0 doesn't use yet.

**Comment by pferraro** [comment]
I see. I had simply cherry-picked the commit from master.  I will fix and re-push.

**Comment by scottmarlow** [comment]
@rvansa could you please review this 5.0 change?  https://github.com/hibernate/hibernate-orm/pull/1269 is similar but for master.

File: hibernate-infinispan/src/main/java/org/hibernate/cache/infinispan/InfinispanRegionFactory.java [hunk 1]
```diff
@@ -431,9 +430,12 @@ public void stop() {
 
 	protected void stopCacheRegions() {
 		log.debug( "Clear region references" );
-		getCacheCommandFactory( manager.getCache().getAdvancedCache() )
-				.clearRegions( regionNames );
-		regionNames.clear();
+		getCacheCommandFactory().clearRegions( regions );
+		// Ensure we cleanup any caches we created
+		for ( BaseRegion region : regions ) {
+			region.getCache().stop();
```
Comment (rvansa) [inline hibernate-infinispan/src/main/java/org/hibernate/cache/infinispan/InfinispanRegionFactory.java]: In the PR to master you undefine the new cache configuration; why not here?

Comment (pferraro) [inline hibernate-infinispan/src/main/java/org/hibernate/cache/infinispan/InfinispanRegionFactory.java]: @rvansa EmbeddedCacheManager.undefineConfiguration(...) wasn't added until Infinispan 8.0.x.  The Hibernate 5.0.x branch requires Infinispan 7.2.x.

Comment (rvansa) [inline hibernate-infinispan/src/main/java/org/hibernate/cache/infinispan/InfinispanRegionFactory.java]: My mistake then, thanks.

**Comment by gbadner** [comment]
Merged. Thanks!

**Comment by heruan** [comment]
I still experience this issue in `5.1`. Is this PR present in any released version?

**Comment by scottmarlow** [comment]
could you try the orm snapshot, as its not in a release yet.  5.1.1-SNAPSHOT

**Comment by heruan** [comment]
I'll try the snapshot, thank you @scottmarlow! Is the next release scheduled yet?

**Comment by scottmarlow** [comment]
Yes @heruan it is in ORM 5.0.9 (you can confirm by checking the HHH-10545, HHH-10546 jiras which should indicate which versions of ORM contain the fix).

**Comment by heruan** [comment]
This bug still affects the new 5.2.0.Final, unfortunately. Is this a regression that should be reported?

**Comment by gbadner** [comment]
@heruan , these commits were applied to master, so it should have been fixed in 5.2.0.Final. If it's not, then please open a new jira with a runnable test case.

**Pull request closed without merging**